app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/recentactivity/dao/RecentActivityDao.kt

```diff
@@ -0,0 +1,31 @@
+package org.wikiedufoundation.wikiedudashboard.ui.coursedetail.recentactivity.dao
+
+import androidx.lifecycle.LiveData
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.recentactivity.data.RecentActivity
+
+/**
+ * This creates an interface for
+ * querying the database
+ **/
+@Dao
+interface RecentActivityDao {
+
+    /**
+     * Get all recent activities
+     *
+     **/
+
+    @Query("SELECT * from recent_activity")
+    fun getRecentActivity(): LiveData<List<RecentActivity>>
```

Comment:
The query will return recentActivity for all courses instead of just 1 course. We need to add where clause in this query for filtering.

Reply:
Which param do I use to query it to get for a specific course? Because there is no foreign key that links the RecentActivity to a course. Maybe like a course_id

Reply:
That is also not possible as course field is not in the recent activity table.

Also, I think offline support for sub-sections of the page is not needed because we will end up storing all data(maybe in GBs) that is stored in backend.

For now I suggest only refactoring to MVVM without offline functionality.

Reply:
Alright.